db/migrations/20201007080108_create_time_ilk_snapshot_query.sql

```diff
@@ -0,0 +1,86 @@
+-- +goose Up
+CREATE INDEX ON api.ilk_snapshot (ilk_identifier, updated);
+
+CREATE TYPE api.time_ilk_snapshot AS
+(
+    bucket_start TIMESTAMP,
+    bucket_end TIMESTAMP,
+    bucket_interval INTERVAL,
+    ilk_identifier TEXT,
+    block_number BIGINT,
+    rate NUMERIC,
+    art NUMERIC,
+    spot NUMERIC,
+    line NUMERIC,
+    dust NUMERIC,
+    chop NUMERIC,
+    lump NUMERIC,
+    dunk NUMERIC,
+    flip TEXT,
+    rho NUMERIC,
+    duty NUMERIC,
+    pip TEXT,
+    mat NUMERIC,
+    created TIMESTAMP,
+    updated TIMESTAMP
+);
+
+-- +goose StatementBegin
+CREATE FUNCTION api.time_ilk_snapshots(ilk_identifier TEXT, range_start TIMESTAMP, range_end TIMESTAMP, bucket_interval INTERVAL DEFAULT '1 day'::INTERVAL)
+    RETURNS SETOF api.time_ilk_snapshot AS
+$$
+DECLARE
+    r api.time_ilk_snapshot%rowtype;
+BEGIN
+    ASSERT EXTRACT(EPOCH FROM (range_end - range_start)) / EXTRACT(EPOCH FROM bucket_interval) <= 1000, 'Please limit requests to at most 1000 buckets.';
```

Comment:
I actually want to increase the amount on the auctions as well. Originally I thought 100 was sufficient, but I've been getting more examples of situations where people are looking for more buckets on mkranalytics.com. I could just code this in on the front end, but the 100 I picked originally was kind of arbitrary, so I figured it would be easier overall to increase it instead. The performance still seems fine even at 1000.

Reply:
👍  cool, that makes sense.

I'd like to take a peak at the current performance of the auction queries before we merge a an increase to the bucket limits, just so we can compare before and after that change is made. But we can get a PR up for that change - feel free to open one yourself, or I can open one later today/tomorrow.

Reply:
Sure, I can take care of that.

Reply:
I've opened a PR: https://github.com/makerdao/vdb-mcd-transformers/pull/393